crates/bevy_ecs/src/world/mod.rs

```diff
@@ -22,15 +22,24 @@ use crate::{
 use std::{
     any::TypeId,
     fmt,
-    sync::atomic::{AtomicU32, Ordering},
+    sync::atomic::{AtomicU32, AtomicU64, Ordering},
 };
 
 #[derive(Copy, Clone, PartialEq, Eq, Debug)]
 pub struct WorldId(u64);
 
+/// The next [`WorldId`]. This will panic on overflow
+static MAX_WORLD_ID: AtomicU64 = AtomicU64::new(0);
+
 impl Default for WorldId {
     fn default() -> Self {
-        WorldId(rand::random())
+        let id = MAX_WORLD_ID
+            // We use `Relaxed` here since this atomic only needs to be consistent with itself
+            .fetch_update(Ordering::Relaxed, Ordering::Relaxed, |val| {
+                val.checked_add(1)
+            })
+            .expect("More bevy `World`s have been created than is supported. Please tell us what you were doing, because frankly I'm impressed.");
```

Comment:
I thought I'd removed this second sentence before committing. I'm happy to keep it as-is though 😆